src/integration/c2b-payment-link/services/binance-pay.service.ts

```diff
@@ -0,0 +1,178 @@
+    const timestamp = Date.now();
+    const nonce = this.getNonce();
+    const signature = this.generateSignature(timestamp, nonce, JSON.stringify(body));
+
+    return {
+      'BinancePay-Timestamp': timestamp,
+      'BinancePay-Nonce': nonce,
+      'BinancePay-Certificate-SN': this.apiKey,
+      'BinancePay-Signature': signature,
+    };
+  }
+
+  public isAvailable(paymentLink: PaymentLinkPayment): boolean {
+    return Boolean(
+      paymentLink.link.configObj.binancePayMerchantId || paymentLink.link.configObj.binancePaySubMerchantId,
+    );
+  }
+
+  async createOrder(
+    payment: PaymentLinkPayment,
+    transferInfo: TransferInfo,
+    quote: PaymentQuote,
+  ): Promise<OrderResult> {
+    const orderDetails: OrderData = {
+      env: {
+        terminalType: 'OTHERS',
+      },
+      merchantTradeNo: quote.uniqueId.replace('plq_', ''),
+      orderAmount: transferInfo.amount,
+      currency: transferInfo.asset,
+      description: payment.memo,
+      goodsDetails: [
+        {
+          goodsType: '01',
+          goodsCategory: 'D000',
+          referenceGoodsId: '01',
+          goodsName: payment.memo,
+          goodsDetail: payment.memo,
+        },
+      ],
+    };
+
+    const { binancePayMerchantId, binancePaySubMerchantId } = payment.link.configObj;
+    if (binancePayMerchantId) {
+      (orderDetails as ChannelPartnerOrderData).merchantId = binancePayMerchantId;
+    } else {
+      (orderDetails as SubMerchantOrderData).merchant = {
+        subMerchantId: binancePaySubMerchantId,
+      };
+    }
+
+    try {
+      const response = await this.http.post<OrderResponse>(
+        `${this.baseUrl}/binancepay/openapi/v3/order`,
+        orderDetails,
+        {
+          headers: this.getHeaders(orderDetails),
+        },
+      );
+
+      const {
+        data: { deeplink, prepayId },
+      } = response;
+
+      return {
+        providerOrderId: prepayId,
+        paymentRequest: deeplink,
+        metadata: response.data,
+      };
+    } catch (error) {
+      throw error.response?.data || error;
+    }
+  }
+
+  async queryCertificate(): Promise<CertificateResponse> {
+    if (this.certificatedExpiry > Date.now()) {
+      return {
+        status: 'SUCCESS',
+        code: '000000',
+        data: this.cert,
+      };
+    }
+
+    this.certificatedExpiry = Date.now() + 5 * 60 * 1000;
+    const headers = this.getHeaders({});
+    const response = await this.http.post<CertificateResponse>(
+      `${this.baseUrl}/binancepay/openapi/certificates`,
+      {},
+      { headers },
+    );
+
+    this.cert = response.data;
+    return response;
+  }
+
+  public async verifySignature(body: BinancePayWebhookDto, headers: BinancePayHeaders): Promise<boolean> {
+    const { timestamp, nonce, signature, certSN } = headers;
+    const webhookData = JSON.stringify({ ...body, bizId: body.bizIdStr }).replace(/"bizId":"(\d+)"/g, '"bizId":$1');
+    const payload = `${timestamp}\n${nonce}\n${webhookData}\n`;
```

Comment:
Why not just `const webhookData = JSON.stringify({ ...body, bizId: +body.bizIdStr });`? 🤔